[Woo POS][Historical Orders] Design Updates: Larger font sizes & Badge Colors & Error icon

staskus opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses. -->
<!-- Part of: # -> use this if your PR is one of many related to one issue -->

## Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

WOOMOB-1420

Design Updates:
- Larger font sizes for various elements: header, totals, products.
- Badge background color
- X error icon for error within the list

## Steps to reproduce
<!-- Step-by-step testing instructions. For new user flows, consider instead stating the goal of the workflow and see if your PR reviewer can accomplish the workflow without specific steps! -->

POS -> Menu -> Orders, confirm:
- Larger font sizes in Order Details
- Different completed and failed badge colors
- X error icon when error happens (no internet connection when orders are loading)

## Testing information
<!-- This is your opportunity to break out individual scenarios that need testing (when necessary) and/or include a checklist for the reviewer to go through. Consider documenting the following from your own completed testing: devices used, alternate workflows, edge cases, affected areas, critical flows, areas not tested, and any remaining unknowns. Provide feedback on this new section of the PR template through Sept 30, 2024 to Apps Quality; additional context here: p91TBi-b8z-p2#comment-12036 -->

Tested don iPadOS 26 simulator

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->

### Updated font sizes & Badge color

https://github.com/user-attachments/assets/4c471d29-1ce0-4bff-8134-74e1587be2c7

### X Error icon

<img width="924" height="713" alt="No orders loaded" src="https://github.com/user-attachments/assets/5cb2c4b0-a181-4468-9d9d-1d8f4b94086b" />
<img width="951" height="892" alt="pos error x mark" src="https://github.com/user-attachments/assets/28af8b81-3898-4b44-95bd-912506b7d051" />

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

staskus committed:
Increase font size in a page header in POSOrderDetailsView

staskus committed:
Increase font sizes in POSOrderDetailsView

staskus committed:
Set custom colors for POSOrderBadgeView

staskus committed:
Increase placeholder image size

staskus committed:
Show POSErrorXMark in POSListErrorView

staskus committed:
Increase font sizes in POSOrderDetailsLoadingView

jaclync commented on Modules/Sources/PointOfSale/Presentation/Orders/POSOrderBadgeView.swift:
```diff
@@ -12,26 +12,33 @@ struct POSOrderBadgeView: View {
+        case .failed:
+            return Color(uiColor: .init(red: 36/255, green: 10/255, blue: 10/255, alpha: 1))
+        default:
+            return Color(uiColor: .init(red: 16/255, green: 21/255, blue: 23/255, alpha: 1))
```
nit: just double-checking that these and other RGB colors in the diffs of this file are one-off colors, that don't make sense to be in the POS color palette?

staskus replied:
They are one-off colors but I had a question of this to the design, I'll update in case it needs to be part of the DS.

staskus replied:
New colors added to the design system after consultation 👍

jaclync reviewed (approved):
Tested the 3 scenarios and worked well :shipit: 

One issue that is most likely unrelated to this PR is a crash only in iOS 18 simulators (built with Xcode 26, I tried iPad A16 in iOS 18.4 and 18.5; no crash in iOS 26 simulator):

- POS -> Menu -> Orders
- After the order list is loaded, go to the Settings app (I was trying to change the font size) --> notice that the app crashes

Crash reason: `SwiftUICore/Environment+Objects.swift:34: Fatal error: No Observable object of type POSOrderListModel found. A View.environmentObject(_:) for POSOrderListModel may be missing as an ancestor of this view.`

I don't have an iOS 18 tablet, might be worth verifying in a physical device.


https://github.com/user-attachments/assets/490ad007-99de-445d-a621-905cd1dd5e2b

staskus committed:
Merge branch 'trunk' into woomob-1420-woo-poshistorical-orders-design-updates-larger-font-sizes

staskus commented:
> One issue that is most likely unrelated to this PR is a crash only in iOS 18 simulators (built with Xcode 26, I tried iPad A16 in iOS 18.4 and 18.5; no crash in iOS 26 simulator):

@jaclync, good catch. I'll try to understand why it's the case 🤔

staskus committed:
Explicitly pass the environment to POSOrderListView and POSOrderDetailsView

On iOS 18.5 the app crashes after putting to background when the environment is not explicitly passed

staskus commented:
@jaclync I was able to reproduce it easily. It was enough to background the app. Great catch again!

There's some differences under-the-hood on SwiftUI between versions and for some reason Order List and Details lose their reference to the environment object when the app is put in the background. The fix is to pass the environmentObject explicitly to both orderList and orderDetails, the crashes no longer happen.

staskus committed:
Address warning: Imported module 'WooFoundation' is unused

staskus committed:
Create posBodyXLargeBold POSFontStyle

staskus committed:
Use posBodyXLargeBold in POS Orders

staskus committed:
Replae posBodyXLarge with posBodyXLargeRegular

staskus committed:
Add new status colors for POSOrderBadgeView

- errorLowest
